theano/gof/destroyhandler.py

```diff
@@ -394,6 +394,31 @@ def get_destroyers_of(r):
                 return []
         fgraph.destroyers = get_destroyers_of
 
+        def has_destroyers(protected_list):
+            visited_app_set = set()
+
+            def recursive_destroys_finder(clients_list):
```

Comment:
I would change the interface to have this function take a variable as input. Then it would loop over var.clients itself. I think it is a nicer interface and will help the conversion to the recursion free version.